lib/snapshot/index.js

```diff
@@ -36,15 +37,35 @@ function getVulnerableFunctionsLocations(moduleInfo) {
   });
 
   try {
-    return ast.findAllVulnerableFunctionsInScript(
+    const vulnerableFunctionsFound = ast.findAllVulnerableFunctionsInScript(
       fs.readFileSync(scriptPath), vulnerableFunctionNames);
+    const foundFunctionNames = Object.keys(vulnerableFunctionsFound);
+    validateFoundFunctions(scriptPathFunctions, moduleInfo, vulnerableFunctionNames, foundFunctionNames);
+    return vulnerableFunctionsFound;
   } catch (error) {
     debug(`Error finding vulnerable methods ${vulnerableFunctionNames}` +
       ` in script path ${scriptPath}: ${error}`);
     return {};
   }
 }
 
+function validateFoundFunctions(scriptPathFunctions, moduleInfo, functionsExpected, functionsActual) {
+  const functionsNotFound = functionsExpected.filter(f => !functionsActual.includes(f));
+  if (functionsNotFound && functionsNotFound.length > 0) {
+    const warningEvent = {
+      message: 'instrumentation discrepacy: missing functions from source code',
+      moduleInfo,
+      snapshotInfo: scriptPathFunctions,
+      functionsInfo: {
+        functionsExpected,
```

Comment:
Why not sending `functionsNotFound` alone? Why do we care about the rest?

Reply:
the more context we have the easier we can diagnose the (potential) problems, I think - at least until (if) we discover there's so much data we can make sense of anything. then we'll know better what we're really interested in.